packages/server/src/process_move_service.ts

```diff
@@ -0,0 +1,97 @@
+import { Board, Marker, Winner } from "@mapistry/take-home-challenge-shared";
+import { WinningLines } from "./constants";
+
+export class ProcessMoveService {
+    board: Board;
+
+    constructor(board: Board) {
+        this.board = board;
+    }
+
+    static calculateMinimaxScore(board: Board, depth: number, aiTurn: boolean) {
+        const scoreMap = {
```

Comment:
I don't like the magic keys and magic values here. Again, I tried playing with this a little to reuse the existing types/interfaces but was having difficulty finding the right abstraction. Pragmatically, it might make sense to wait to work on these until more of this feature was developed.